public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -43,13 +44,13 @@ export class PrometheusDatasource {
     this.resultTransformer = new ResultTransformer(templateSrv);
   }
 
-  _request(method, url, data?, requestId?) {
+  _request(url, data?, options?: any) {
     var options: any = {
       url: this.url + url,
-      method: method,
-      requestId: requestId,
+      method: this.httpMethod,
```

Comment:
This does not work for *metric find queries* when using Prometheus >= v2.1.0 and HTTP method is configured to POST. When editing a template variable query I receive "Template variables could not be initialized: Cannot read property 'jquery' of null". 

I think all meta data requests should always use GET to resolve this problem. Some tests of this would be nice for the _request method and/or *metric find queries*.

Reply:
Good catch!